server/hotrod/src/main/scala/org/infinispan/server/hotrod/HotRodServer.scala

```diff
@@ -247,7 +248,19 @@ class HotRodServer extends AbstractProtocolServer("HotRod") with Log {
       clientListenerRegistry.setEventMarshaller(Option(marshaller))
    }
 
-   override def stop: Unit = {
+   def addCacheStore(name: String, instance: Object): Unit = {
```

Comment:
Yes, the deployment order is indeed tricky. I may implement a `Semaphore` there, just to make sure that deployed caches are available when initializing them in `PersistenceManagerImpl` but this solution is not too elegant. Perhaps there is a better one? Is there any way to ensure that deployment processors finished their job before proceeding with Persistence startup?